src/test/java/uk/gov/hmcts/reform/migration/service/ApplicantDataMigrationServiceTest.java

```diff
@@ -0,0 +1,229 @@
+
+    private final ApplicantDataMigrationService service = new ApplicantDataMigrationService();
+
+    @Test
+    void whenOldStructureDoesNotExistAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(ImmutableMap.of("data", "someData"))
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+    @Test
+    void whenDataIsNullAcceptsShouldReturnFalse() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(null)
+            .build();
+
+        assertThat(service.accepts().test(caseDetails)).isEqualTo(false);
+    }
+
+    @Test
+    void whenOldApplicantStructureIsMigratedShouldReturnNewAplicantStructure() {
+        Map<String, Object> data = new HashMap<>();
+
+        OldApplicant applicant = getOldApplicant();
+
+        Applicant newApplicant = newApplicantBuilder();
+
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        service.migrate(caseDetails);
+
+
+        List<Map<String, Object>> valueInApplicant = (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+        assertThat(valueInApplicant.get(0).get("value").equals(newApplicant));
+    }
+
+    @Test
+    void whenPartiallyFilledInOldApplicantStructureIsMigratedShouldReturnNewListStructureWithNullFields() {
+        Map<String, Object> data = new HashMap<>();
+        OldApplicant applicant = OldApplicant.builder()
+            .name(APPLICANT)
+            .email(EMAIL)
+            .build();
+        
+        data.put("applicant", applicant);
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .id(1111L)
+            .data(data)
+            .build();
+
+        service.migrate(caseDetails);
+
+        List<Map<String, Object>> valueInApplicant = (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+        assertThat(valueInApplicant).hasSize(1);
+        assertThat(valueInApplicant.get(0).get("value").equals(Applicant.builder()
+            .party(ApplicantParty.builder()
+                .partyId(null)
+                .partyType(null)
+                .name(APPLICANT)
+                .address(Address.builder()
+                    .build())
+                .emailAddress(EmailAddress.builder()
+                    .email(EMAIL)
+                    .build())
+                .telephoneNumber(TelephoneNumber.builder()
+                    .build())
+                .mobileNumber(MobileNumber.builder()
+                    .build())
+                .jobTitle(null)
+                .pbaNumber(null)
+                .build())
+                .leadApplicantIndicator(null)
+            .build()));
+    }
+
+    @Test
+    void whenPartiallyFilledInOldApplicantAddressIsMigratedShouldReturnNewListStructureWithNullFields() {
```

Comment:
I am not quite sure why we have separate case for postcode and county populated? Is that common practice for cases to have only the two populated?

Reply:
It is to test partial completion of builders like address that the user is still allowed to submit.